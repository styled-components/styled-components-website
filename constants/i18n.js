const DEFAULT_TRANSLATION = 'translations'
const NAV_TRANSLATION = 'nav'
const HOME_TRANSLATION = 'home'
const DOCS_TRANSLATION = 'docs'
const DOCS_BASICS_TRANSLATION = 'docsBasics'
const DOCS_ADVANCED_TRANSLATION = 'docsAdvanced'
const DOCS_API_TRANSLATION = 'docsApi'
const DOCS_FAQS_TRANSLATION = 'docsFaqs'

const TRANSLATIONS = [
  DEFAULT_TRANSLATION,
  NAV_TRANSLATION,
  HOME_TRANSLATION,
  DOCS_TRANSLATION,
  DOCS_BASICS_TRANSLATION,
  DOCS_ADVANCED_TRANSLATION,
  DOCS_API_TRANSLATION,
  DOCS_FAQS_TRANSLATION,
]

const DEFAULT_LANGUAGE = 'en'

const LANGUAGES = [
  DEFAULT_LANGUAGE,
  'es',
]

module.exports = {
  DEFAULT_TRANSLATION,
  NAV_TRANSLATION,
  TRANSLATIONS,
  HOME_TRANSLATION,
  DOCS_TRANSLATION,
  DOCS_BASICS_TRANSLATION,
  DOCS_ADVANCED_TRANSLATION,
  DOCS_API_TRANSLATION,
  DOCS_FAQS_TRANSLATION,
  LANGUAGES,
  DEFAULT_LANGUAGE,
}
