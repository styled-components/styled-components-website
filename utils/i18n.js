import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { reactI18nextModule } from 'react-i18next'

import translations from '../locales'

i18n
  .use(LanguageDetector)
  .use(reactI18nextModule)
  .init({
    fallbackLng: 'en',

    // have a common namespace used around the full app
    ns: ['translations', 'homepage'],
    defaultNS: 'translations',

    debug: process.env.NODE_ENV !== 'production',

    interpolation: {
      escapeValue: false, // not needed for react!!
    },

    react: {
      wait: true
    },

    resources: { ...translations }
  })

export default i18n
