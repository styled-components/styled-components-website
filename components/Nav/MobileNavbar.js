import React from 'react';
import styled, { css } from 'styled-components';
import { Search, KeyboardArrowDown } from '@styled-icons/material';
import rem from '../../utils/rem';
import { navbarHeight } from '../../utils/sizes';
import { paleGrey } from '../../utils/colors';
import { mobile } from '../../utils/media';
import { CloseIcon, FoldIcon } from './NavIcons';
import Link from '../Link';
import NavLinks from './NavLinks';
import Social from './Social';
import Logo from './Logo';
import NavButton from './NavButton';

const Wrapper = styled.div`
  display: none;

  ${mobile(css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: ${rem(navbarHeight)};
  `)};
`;

const SecondaryMenu = styled.div`
  position: absolute;
  top: ${rem(navbarHeight)};
  left: 0;
  right: 0;

  ${p =>
    p.isOpen
      ? css`
          height: ${rem(navbarHeight)};
        `
      : css`
          height: 0;
        `} display: flex;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  padding: 0 ${rem(20)};
  transition: height 0.1s;

  user-select: none;
  -webkit-overflow-scrolling: touch;
  overflow-x: scroll;
  overflow-y: hidden;

  background: ${paleGrey};
  color: #868686;
`;

const LogoLink = styled(Link).attrs((/* props */) => ({
  unstyled: true,
  href: '/',
  'aria-label': 'styled components',
}))`
  display: inline-block;
  vertical-align: center;
`;

const ArrowWrapper = styled.div`
  transition: transform 0.2s;

  ${p =>
    p.shouldRotate &&
    css`
      transform-origin: center center;
      transform: rotate(180deg);
    `};
`;

const SecondaryMenuItem = styled.div`
  padding-right: ${rem(20)};
`;

const StyledIcon = styled.div`
  && {
    width: ${p => rem(p.size || 20)};
    height: ${p => rem(p.size || 20)};
  }
`;

const MobileNavbar = props => {
  const { isSideFolded, isMobileNavFolded, onSideToggle, onMobileNavToggle, showSideNav, onSearchButtonClick } = props;

  return (
    <Wrapper>
      {showSideNav !== false && (
        <NavButton active={!isSideFolded} onClick={onSideToggle}>
          {isSideFolded ? <FoldIcon /> : <CloseIcon />}
        </NavButton>
      )}

      <LogoLink>
        <Logo compact />
      </LogoLink>
      <div>
        <NavButton onClick={onSearchButtonClick}>
          <StyledIcon as={Search} size={28} />
        </NavButton>

        <NavButton onClick={onMobileNavToggle} active={!isMobileNavFolded}>
          <ArrowWrapper shouldRotate={!isMobileNavFolded}>
            <StyledIcon as={KeyboardArrowDown} size={36} />
          </ArrowWrapper>
        </NavButton>
      </div>

      <SecondaryMenu isOpen={!isMobileNavFolded}>
        <NavLinks />
        <SecondaryMenuItem>
          <Social />
        </SecondaryMenuItem>
      </SecondaryMenu>
    </Wrapper>
  );
};

export default MobileNavbar;
