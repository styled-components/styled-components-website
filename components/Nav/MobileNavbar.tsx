import { KeyboardArrowDown } from '@styled-icons/material';
import styled, { css } from 'styled-components';
import { paleGrey } from '../../utils/colors';
import { mobile } from '../../utils/media';
import rem from '../../utils/rem';
import { navbarHeight } from '../../utils/sizes';
import Link from '../Link';
import Logo from './Logo';
import NavButton from './NavButton';
import { CloseIcon, FoldIcon } from './NavIcons';
import NavLinks from './NavLinks';
import Social from './Social';

const Wrapper = styled.div`
  display: none;

  ${mobile(css`
    align-items: center;
    display: flex;
    flex-grow: 1;
    height: ${rem(navbarHeight)};
    justify-content: space-between;
  `)};
`;

const SecondaryMenu = styled.div<{ $isOpen?: boolean }>`
  position: absolute;
  top: ${rem(navbarHeight)};
  left: 0;
  right: 0;

  ${p =>
    p.$isOpen
      ? css`
          height: ${rem(navbarHeight)};
        `
      : css`
          height: 0;
        `} display: flex;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  padding: 0 ${rem(20)};
  transition: height 0.1s;

  user-select: none;
  -webkit-overflow-scrolling: touch;
  overflow-x: scroll;
  overflow-y: hidden;

  background: ${paleGrey};
  color: #868686;
`;

const LogoLink = styled(Link).attrs((/* props */) => ({
  unstyled: true,
  href: '/',
  'aria-label': 'styled components',
}))`
  display: inline-block;
  vertical-align: center;
`;

const ArrowWrapper = styled.div<{ $shouldRotate?: boolean }>`
  transition: transform 0.2s;

  ${p =>
    p.$shouldRotate &&
    css`
      transform-origin: center center;
      transform: rotate(180deg);
    `};
`;

const SecondaryMenuItem = styled.div`
  padding-right: ${rem(20)};
`;

const StyledIcon = styled.div<{ $size?: number }>`
  && {
    width: ${p => rem(p.$size || 20)};
    height: ${p => rem(p.$size || 20)};
  }
`;

export interface MobileNavbarProps {
  isMobileNavFolded?: boolean;
  isSideFolded?: boolean;
  onMobileNavToggle?: () => void;
  onSideToggle?: () => void;
  showSideNav?: boolean;
}

const MobileNavbar = ({
  children,
  isMobileNavFolded,
  isSideFolded,
  onMobileNavToggle,
  onSideToggle,
  showSideNav,
}: React.PropsWithChildren<MobileNavbarProps>) => {
  return (
    <Wrapper>
      {showSideNav !== false && (
        <NavButton $active={!isSideFolded} onClick={onSideToggle}>
          {isSideFolded ? <FoldIcon /> : <CloseIcon />}
        </NavButton>
      )}

      <LogoLink>
        <Logo $compact />
      </LogoLink>

      {children}

      <NavButton onClick={onMobileNavToggle} $active={!isMobileNavFolded} style={{ position: 'absolute', right: 0 }}>
        <ArrowWrapper $shouldRotate={!isMobileNavFolded}>
          <StyledIcon as={KeyboardArrowDown} $size={36} />
        </ArrowWrapper>
      </NavButton>

      <SecondaryMenu $isOpen={!isMobileNavFolded}>
        <NavLinks />
        <SecondaryMenuItem>
          <Social />
        </SecondaryMenuItem>
      </SecondaryMenu>
    </Wrapper>
  );
};

export default MobileNavbar;
