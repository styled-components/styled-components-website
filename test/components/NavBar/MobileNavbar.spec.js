import React from 'react'
import { mount } from 'enzyme'
import 'jest-styled-components'
import { I18nextProvider } from 'react-i18next'
import MobileNavbar from '../../../components/Nav/MobileNavbar'

import i18n from '../../helpers/i18n'

const wrapper = (language, props) => mount(
  <I18nextProvider i18n={i18n} initialLanguage={language}>
    <MobileNavbar />
  </I18nextProvider>
)

test('MobileNavbar renders correctly', () => {
  const component = wrapper('en')
  expect(component.find(MobileNavbar)).toMatchSnapshot()
})
