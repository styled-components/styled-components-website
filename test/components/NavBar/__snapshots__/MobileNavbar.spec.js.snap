// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNavbar renders correctly 1`] = `
.c3 svg {
  display: inline-block;
}

.c3 svg path {
  fill: currentColor;
}

.c13 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 0.2777777777777778rem;
  height: 0.2777777777777778rem;
  margin: 0 0.8333333333333334rem;
  border-radius: 50%;
  background: currentColor;
  opacity: 0.35;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-right: 1.6666666666666667rem;
}

.c12 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: inline-block;
  line-height: 2.7777777777777777rem;
  -webkit-transition: opacity 0.2s,-webkit-transform 0.2s;
  -webkit-transition: opacity 0.2s,transform 0.2s;
  transition: opacity 0.2s,transform 0.2s;
  cursor: pointer;
  -webkit-letter-spacing: 0.022222222222222223rem;
  -moz-letter-spacing: 0.022222222222222223rem;
  -ms-letter-spacing: 0.022222222222222223rem;
  letter-spacing: 0.022222222222222223rem;
  color: currentColor;
}

.c12:hover,
.c12:focus {
  opacity: 0.8;
}

.c12:active {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
  opacity: 0.6;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: 1.1111111111111112rem;
  line-height: 2.7777777777777777rem;
  -webkit-transition: opacity 0.2s,-webkit-transform 0.2s;
  -webkit-transition: opacity 0.2s,transform 0.2s;
  transition: opacity 0.2s,transform 0.2s;
  cursor: pointer;
}

.c16:last-child {
  margin-right: 0;
}

.c16:hover,
.c16:focus {
  opacity: 0.8;
}

.c16:active {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
  opacity: 0.6;
}

.c16 svg path {
  fill: currentColor;
}

.c17 {
  width: 0.7777777777777778rem;
  height: 0.7777777777777778rem;
}

.c18.c18 {
  width: NaNrem;
  height: 1rem;
}

.c5 {
  display: inline-block;
  vertical-align: middle;
  box-sizing: border-box;
  background-position: center;
  background-size: contain;
  background-image: url(/icon.png);
  width: 5.555555555555555rem;
  height: 1.6666666666666667rem;
}

.c2 {
  background: none;
  outline: none;
  border: none;
  background: rgba(0,0,0,0.07);
  cursor: pointer;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 2.7777777777777777rem;
  padding: 0 0.5555555555555556rem;
  text-align: center;
  vertical-align: middle;
}

.c1 + .c1 {
  padding-left: 0;
}

.c6 {
  background: none;
  outline: none;
  border: none;
  background: none;
  cursor: pointer;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 2.7777777777777777rem;
  padding: 0 0.5555555555555556rem;
  text-align: center;
  vertical-align: middle;
}

.c1 + .c1 {
  padding-left: 0;
}

.c0 {
  display: none;
}

.c10 {
  position: absolute;
  top: 2.7777777777777777rem;
  left: 0;
  right: 0;
  height: 2.7777777777777777rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 1.1111111111111112rem;
  -webkit-transition: height 0.1s;
  transition: height 0.1s;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-overflow-scrolling: touch;
  overflow-x: scroll;
  overflow-y: hidden;
  background: #efefef;
  color: #868686;
}

.c4 {
  display: inline-block;
  vertical-align: center;
}

.c8 {
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  -webkit-transform-origin: center center;
  -ms-transform-origin: center center;
  transform-origin: center center;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c14 {
  padding-right: 1.1111111111111112rem;
}

.c7.c7 {
  width: 1.5555555555555556rem;
  height: 1.5555555555555556rem;
}

.c9.c9 {
  width: 2rem;
  height: 2rem;
}

@media (max-width:62.5em) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    height: 2.7777777777777777rem;
  }
}

<div
  className="c0"
>
  <button
    className="c1 c2"
  >
    <svg
      className="c3"
      height="15"
      width="15"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <title>
        close
      </title>
      <use
        fill="#FFF"
        transform="translate(1 1)"
        xlinkHref="#close"
      />
      <defs>
        <path
          d="M-.7.7l13 13 1.4-1.4-13-13L-.7.7zm13-1.4l-13 13 1.4 1.4 13-13-1.4-1.4z"
          id="close"
        />
      </defs>
    </svg>
  </button>
  <a
    aria-label="styled components"
    className="c4"
    href="/"
  >
    <div
      className="c5"
    />
  </a>
  <div>
    <button
      className="c1 c6"
    >
      <svg
        className="c7"
        size={28}
      />
    </button>
    <button
      className="c1 c2"
    >
      <div
        className="c8"
      >
        <div
          className="c9"
          size={36}
        />
      </div>
    </button>
  </div>
  <div
    className="c10"
  >
    <nav
      className="c11"
    >
      <a
        className="c12"
        href="/docs"
      >
        Documentation
      </a>
      <span
        className="c13"
      />
      <a
        className="c12"
        href="/showcase"
      >
        Showcase
      </a>
      <span
        className="c13"
      />
      <a
        className="c12"
        href="/ecosystem"
      >
        Ecosystem
      </a>
      <span
        className="c13"
      />
      <a
        className="c12"
        href="/releases"
      >
        Releases
      </a>
    </nav>
    <div
      className="c14"
    >
      <nav
        className="c15"
      >
        <a
          className="c16"
          href="https://spectrum.chat/styled-components/"
        >
          <svg
            className="c17"
            height="14"
            viewBox="0 0 15 15"
            width="14"
          >
            <title>
              spectrum
            </title>
            <path
              d="M0 6.5V1c0-.6.4-1 1-1 9 .3 13.7 5 14 14 0 .6-.4 1-1 1H8.5c-.6 0-1-.4-1-1-.3-4.4-2-6.2-6.5-6.5-.6 0-1-.4-1-1z"
              fill="#FFF"
            />
          </svg>
        </a>
        <a
          className="c16"
          href="https://github.com/styled-components"
        >
          <svg
            className="c18"
            height="18"
          />
        </a>
        <a
          className="c16"
          href="https://medium.com/styled-components"
        >
          <svg
            className="c18"
            height="18"
          />
        </a>
      </nav>
    </div>
  </div>
</div>
`;
